### PR TITLE
Don't allow to create subprojects with same alias

### LINKED ISSUE #5402: Don't allow to create subprojects with same alias
Currently we can create several subprojects with the same alias, when serving the docs from 

https://github.com/rtfd/readthedocs.org/blob/bdb4aa5da5b5b5fb45244176c2bee58fb5442414/readthedocs/core/views/serve.py#L71-L74

It gives a 500 error.

### PR DESCRIPTION
**Pull request by saadmk11**
Added a `clean_alias` method in the `ProjectRelationshipBaseForm` which will check if the alias already exists. This will prevent Projects from Creating New subprojects with same alias. 
closes #5402

### CODE REVIEW

File: readthedocs/projects/forms.py [hunk 1]
```diff
@@ -363,6 +363,17 @@ def clean_child(self):
...
         return child
 
+    def clean_alias(self):
+        alias = self.cleaned_data['alias']
+        subproject = self.project.subprojects.filter(
+            alias=alias).exclude(id=self.instance.pk)
+
+        if subproject.exists():
+            raise forms.ValidationError(
+                _('A subprojects with this alias already exists'),
```
Comment (stsewd) [inline readthedocs/projects/forms.py]:
```suggestion
                _('A subproject with this alias already exists'),
```

**Comment by saadmk11** [comment]
@stsewd Okay. I'll add some tests and update the PR.

**Comment by saadmk11** [comment]
@stsewd I have Added some Tests and Updated the PR.

**Comment by saadmk11** [comment]
@stsewd Thanks for pointing it out. I couldn’t  find the existing test case as the name is different. I'll move the tests here and update the PR.

**Comment by saadmk11** [comment]
@stsewd Moved the tests to `test_subprojects.py`

**Comment by saadmk11** [comment]
@stsewd Should I remove the tests that are already there or this can be merged?

**Comment by stsewd** [comment]
You can try to identify the repeated tests, but it's fine for me.

**Comment by saadmk11** [comment]
@stsewd okay cool.

**Comment by saadmk11** [comment]
@stsewd Removed Repeated tests. :)

**Comment by stsewd** [comment]
Tested this locally, it's working

**Pull request merged**